### PR TITLE
Kube prometheus stack upgrade

### LINKED ISSUE #664: [3] Upgrade kube-prometheus-stack to v19.2.2
**What Chart should be upgraded?**
upstream / kube-prometheus-stack: [prometheus-community/kube-prometheus-stack](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack)
v0.48.1/v16.6.1 (2021/06/10) -> v0.50.0/v19.2.2 (2021/10/19)

**Subcharts**
- grafana 8.0.0 -> v8.2.2/v6.17.4 (2021/10/27)
  - Also update upstream / grafana: [grafana/grafana](https://github.com/grafana/helm-charts/tree/main/charts/grafana)
- kube-state-metrics v2.0.0/v3.1.1 -> v2.2.0/v3.5.2
- prometheus v2.27.1 -> v2.28.1
- prometheus-node-exporter v1.1.2/v1.18.1 -> v1.2.2/v2.2.0

**Why should this Chart be upgraded?**

- [ ] We need to perform a security upgrade.
- [ ] We need it for a new feature: *which feature (link to blocked issue)*.
- [x] We need to keep up to prevent future issues.
- [ ] Other: *Write the reason here*

Need to upgrade this for the operator and kube-state-metrics to support APIs on K8s v1.22 going forward.

Also update chart / prometheus-instance from using deprecated ingress API version.

**Acceptance criteria**

- I checked the migration of the new Chart:
    - [ ] I upgraded a Chart and determined that no migration steps are needed.
    - [ ] I upgraded a Chart and added [migration steps](https://github.com/elastisys/compliantkubernetes-apps/blob/main/migration).
- [ ] I tested the functionality provided by the new Chart (e.g., Auth flow, Grafana dashboards, etc.)

### PR DESCRIPTION
**Pull request by crssnd**
**What this PR does / why we need it**: Need to upgrade this for the operator and kube-state-metrics to support APIs on K8s v1.22 going forward.

**Which issue this PR fixes** *(use the format `fixes #<issue number>(, fixes #<issue_number>, ...)` to automatically close the issue when PR gets merged)*: fixes #664 

**Public facing documentation PR** *(if applicable)*

**Special notes for reviewer**: 

**Checklist:**

- [x] Added relevant notes to [WIP-CHANGELOG.md](https://github.com/elastisys/compliantkubernetes-apps/blob/main/WIP-CHANGELOG.md)
- [x] Proper commit message prefix on all commits
- [ ] Updated the [public facing documentation](https://github.com/elastisys/compliantkubernetes)
- Is this changeset backwards compatible for existing clusters? Applying:
    - [ ] is completely transparent, will not impact the workload in any way.
    - [ ] requires running a migration script.
    - [x] will create noticeable cluster degradation.
          E.g. logs or metrics are not being collected or Kubernetes API server
          will not be responding while upgrading.
    - [ ] requires draining and/or replacing nodes.
    - [ ] will change any APIs.
          E.g. removes or changes any CK8S config options or Kubernetes APIs.
    - [ ] will break the cluster.
          I.e. full cluster migration is required.
- Chart checklist (pick exactly one):
    - [ ] I upgraded no Chart.
    - [ ] I upgraded a Chart and determined that no migration steps are needed.
    - [x] I upgraded a Chart and added [migration steps](https://github.com/elastisys/compliantkubernetes-apps/blob/main/migration).

**Pipeline config** *(if applicable)*
If you change some config options (e.g. add/rename variable or change the default value) you may need to update the config used by the pipeline in `pipeline/config`.

### CODE REVIEW

**Review by Xartos — APPROVED** [review]
LGTM

**Review by Ajarmar — APPROVED** [review]
LGTM, although it seems like this PR also includes all the changes from https://github.com/elastisys/compliantkubernetes-apps/pull/681. Just make sure to merge that one first.

File: helmfile/values/prometheus-alerts-sc.yaml.gotmpl [hunk 1]
```diff
@@ -1,4 +1,7 @@
 esNodeCount: {{ add .Values.elasticsearch.dataNode.count .Values.elasticsearch.clientNode.count .Values.elasticsearch.masterNode.count }}
+eskubeAuditSizeGB: {{ .Values.elasticsearch.curator.retention.kubeAuditSizeGB }}
+eskubernetesSizeGB: {{ .Values.elasticsearch.curator.retention.kubernetesSizeGB }}
+esotherSizeGB: {{ .Values.elasticsearch.curator.retention.otherSizeGB }}
```
Comment (viktor-f) [inline helmfile/values/prometheus-alerts-sc.yaml.gotmpl:4]: Should you have one for `kubeaudit` as well?

File: WIP-CHANGELOG.md [hunk 2]
```diff
@@ -4,16 +4,69 @@
 
 # Updated
 
- - updated falco helm chart to version 1.16.0, this upgrades falco to version 0.30.0
+- updated falco helm chart to version 1.16.0, this upgrades falco to version 0.30.0
+- kube-prometheus-stack to v19.2.2 [#685](https://github.com/elastisys/compliantkubernetes-apps/pull/685)
```
Comment (OlleLarsson) [inline WIP-CHANGELOG.md:8]: We had a short discussion offline regarding this changelog entry and we concluded that it should be reduced, to only include that of which is important to our product. Please keep the link to the full changelogs though.

File: helmfile/values/prometheus-alerts-sc.yaml.gotmpl (hunk 1 above)
Comment (crssnd) [reply helmfile/values/prometheus-alerts-sc.yaml.gotmpl:4]: done

File: WIP-CHANGELOG.md (hunk 2 above)
Comment (crssnd) [reply WIP-CHANGELOG.md:8]: done please check.

**Pull request merged**